### PR TITLE
Fix dashboard subscriptions sidebar for sandboxed users

### LINKED ISSUE #17446: Sandboxes users and Dashboard Subscription shows Slack incorrectly
**Describe the bug**
When a user is Sandboxed, then viewing Dashboard Subscriptions will show Slack as not configured. And if admin has created a Slack subscription, then the Sandboxes user can see some details of that, but cannot edit/view the subscription.

Regression since 1.38.0 or more precisely 1.38.3 (#14990)

Slack is not available for Sandboxed users (for historic reasons) - #14844

**To Reproduce**
1. As admin, setup Slack and create a Slack subscription on a dashboard
image
2. Login as Sandboxed user, go to the dashboard and open the Subscription sidebar - shows subscription, but it's not possible to click the subscription (frontend reloads with `TypeError: a is undefined` and `COMPONENT DID CATCH LOLE`) - it should probably not show any Slack subscriptions?
image
3. As Sandboxed, clicking the :heavy_plus_sign: will show Slack as not configured (it should not be shown to non-admins)
image

**Expected behavior**
It's getting difficult to figure out what the intended behavior should be, but I would guess not show anything Slack related to Sandboxed users (at least until #14844 is implemented).

**Information about your Metabase Installation:**
Tested 1.38.3 thru 1.40.2

### PR DESCRIPTION
**Pull request by ranquild**
Fixes https://github.com/metabase/metabase/issues/17446.

How to test:
- Follow the steps from the linked ticket
- When viewing existing subscriptions for a dashboard, sandbox users should not be able to edit slack channels

### CODE REVIEW

File: frontend/src/metabase/sharing/components/NewPulseSidebar.jsx [hunk 1]
```diff
@@ -23,91 +24,104 @@ function NewPulseSidebar({
...
-                  {
-                    "text-brand hover-child hover--inherit": emailConfigured,
-                  },
-                  { "text-light": !emailConfigured },
-                )}
-              />
-              <h3
-                className={cx({ "text-light": !emailConfigured })}
-              >{t`Email it`}</h3>
+        {(emailConfigured || isAdmin) && (
```
Comment (ranquild) [inline frontend/src/metabase/sharing/components/NewPulseSidebar.jsx:27]: This is the only new line in this block

File: frontend/src/metabase/sharing/components/NewPulseSidebar.jsx [hunk 2]
```diff
@@ -23,91 +24,104 @@ function NewPulseSidebar({
...
-          <div className="px3 pt3 pb2">
-            <div className="flex align-center mb1">
-              <Icon
-                name={slackConfigured ? "slack_colorized" : "slack"}
-                size={24}
-                className={cx("mr1", {
-                  "text-light": !slackConfigured,
+          </Card>
+        )}
+        {(slackConfigured || isAdmin) && (
```
Comment (ranquild) [inline frontend/src/metabase/sharing/components/NewPulseSidebar.jsx:73]: This is the only new line in this block

**Comment by rlotun** [comment]
is it possible to add a repro cypress test for this too please?

**Review by alxnddr — APPROVED** [review]
👍

**Pull request merged**